### PR TITLE
chore: update python versions in CI

### PR DESCRIPTION
**Pull request by cjrh**
Python 3.8 is no longer available on the "macos-latest" target.  This is the current list of supported Python versions: https://devguide.python.org/versions/

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import nox
 
 
-@nox.session(python=["3.8", "3.9", "3.10", "3.11", "3.12"])
+@nox.session(python=["3.8", "3.9", "3.10", "3.11", "3.12", "3.13"])
```
Comment (wallies) [inline noxfile.py:4]: Shouldn't we remove 3.8 from here

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -40,10 +40,11 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        python-version: [3.9]
+        python-version: ["3.12"]
```
Comment (wallies) [inline .github/workflows/ci.yml:43]: Why this change, aren't we removing 3.8?

Comment (cjrh) [reply .github/workflows/ci.yml:43]: - 3.12 is the current stable version, so that's the one we should test across platforms
- 3.8 is still supported by upstream

Comment (cjrh) [reply .github/workflows/ci.yml:43]: image

**Review by wallies — APPROVED** [review]
